src/porepy/numerics/ad/equation_system.py

```diff
@@ -399,57 +399,69 @@ def create_variables(
             non_admissible = requested_type.difference(self.admissible_dof_types)
             raise ValueError(f"Non-admissible DOF types {non_admissible} requested.")
 
-        # Sanity check if variable is defined anywhere.
-        if subdomains is None and interfaces is None:
-            raise ValueError(
-                "Cannot create variable not defined on any subdomain or interface."
-            )
-
         # Container for all grid variables.
         variables = []
 
         # Merge subdomains and interfaces into a single list.
-        assert subdomains is not None or interfaces is not None  # for mypy
-        grids: Sequence[pp.GridLike] = subdomains if subdomains else interfaces  # type: ignore
+        grids: Sequence[pp.GridLike]
+        if subdomains is not None and interfaces is None:
+            grids = subdomains
+        elif subdomains is None and interfaces is not None:
+            grids = interfaces
+        elif subdomains is None and interfaces is None:
+            raise ValueError(
+                "Cannot create variable not defined on any subdomain or interface."
+            )
+        else:
+            # Yura: The current behavior is just to ignore the interfaces, which is
+            # confusing for me. If this usage is not expected, should we throw an error?
+            grids = subdomains + interfaces  # type: ignore
 
         # Check if a md variable was already defined under that name on any of grids.
         for var in self.variables:
             if var.name == name and var.domain in grids:
                 raise KeyError(f"Variable {name} already defined on {var.domain}.")
 
-        if grids:
-            for grid in grids:
-                if subdomains:
-                    assert isinstance(grid, pp.Grid)  # mypy
-                    data = self.mdg.subdomain_data(grid)
-                else:
-                    assert isinstance(grid, pp.MortarGrid)  # mypy
-                    data = self.mdg.interface_data(grid)
+        def initialize_boundary_data(sd: pp.Grid):
```

Comment:
I'm not convinced this is the right place to initialize boundary data. Let's discuss!

Reply:
We decided to keep it here for now during the discussion.